packages/diagrams/backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/ELKDiagramConverter.java

```diff
@@ -288,4 +292,87 @@ private void convertEdge(Edge edge, ElkNode elkDiagram, Map<String, ElkConnectab
+        rightPadding = nodeLabelsPadding.getRight();
+        topPadding = nodeLabelsPadding.getTop();
+        bottomPadding = nodeLabelsPadding.getBottom();
+        maxPadding = Double.max(Double.max(leftPadding, rightPadding), Double.max(topPadding, bottomPadding));
+
+        return maxPadding;
+    }
+
+    private String getNodeLabelType(Node node, ISiriusWebLayoutConfigurator layoutConfigurator) {
+        String labelType = "label:";
+        String nodeType = node.getType();
+        EnumSet<NodeLabelPlacement> nodeLabelPlacementSet = layoutConfigurator.configureByType(nodeType).getProperty(CoreOptions.NODE_LABELS_PLACEMENT);
+        if (nodeLabelPlacementSet.stream().anyMatch(NodeLabelPlacement.OUTSIDE::equals)) {
+            labelType += "outside";
+            if (NodeType.NODE_IMAGE.equals(nodeType)) {
+                labelType += "-center";
+            }
+        } else if (NodeType.NODE_RECTANGLE.equals(nodeType) && !this.hasHeader(node)) {
+            labelType += "inside";
+            String verticalAlignment = this.getVerticalAlignment(nodeLabelPlacementSet);
+            labelType += verticalAlignment;
+
+            String horizontalAlignment = this.getHorizontalAlignment(nodeLabelPlacementSet);
+            labelType += horizontalAlignment;
+        } else {
+            labelType += "inside-center";
+        }
+
+        return labelType;
+    }
+
+    private String getVerticalAlignment(EnumSet<NodeLabelPlacement> nodeLabelPlacementSet) {
+        String verticalAlignment = "-v_center";
+        for (NodeLabelPlacement nodeLabelPlacement : nodeLabelPlacementSet) {
+            switch (nodeLabelPlacement) {
+                case V_TOP:
+                    verticalAlignment = "-v_top";
+                    break;
+                case V_BOTTOM:
+                    verticalAlignment = "-v_bottom";
+                    break;
+                default:
+                    break;
+            }
+        }
+        return verticalAlignment;
+    }
+
+    private String getHorizontalAlignment(EnumSet<NodeLabelPlacement> nodeLabelPlacementSet) {
+        String horizontalAlignment = "-h_center";
+        for (NodeLabelPlacement nodeLabelPlacement : nodeLabelPlacementSet) {
+            switch (nodeLabelPlacement) {
+                case H_LEFT:
+                    horizontalAlignment = "-h_left";
+                    break;
+                case H_RIGHT:
+                    horizontalAlignment = "-h_right";
+                    break;
+                default:
+                    break;
+            }
+        }
+        return horizontalAlignment;
+    }
+
+    private boolean hasHeader(Node node) {
+        INodeStyle nodeStyle = node.getStyle();
+        if (nodeStyle instanceof RectangularNodeStyle) {
+            RectangularNodeStyle rectangularNodeStyle = (RectangularNodeStyle) nodeStyle;
+            return rectangularNodeStyle.isWithHeader();
+        }
+        return false;
```

Comment:
Mor style issue: with Java 17 this can now be:
```java
return node.getStyle() instanceof RectangularNodeStyle rectangularNodeStyle && rectangularNodeStyle.isWithHeader();
```

Reply:
Done